lib/metadata/in_memory/backend.js

```diff
@@ -41,7 +42,8 @@ const metastore = {
             if (!metadata.buckets.has(bucketName)) {
                 return cb(errors.NoSuchBucket);
             }
-            return cb(null, metadata.buckets.get(bucketName));
+            return cb(null, BucketInfo.fromObj(
```

Comment:
@rahulreddy i take a look at the backward PR, if you look, the previous state was to directly do new BucketInfo(), i just replace it by the function fromObj, so its not related to the backward compatibility :)